emop-maven-plugin/src/main/java/edu/cornell/RppMojo.java

```diff
@@ -0,0 +1,88 @@
+package edu.cornell;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.lang.reflect.InvocationTargetException;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+
+import edu.cornell.emop.maven.SurefireMojoInterceptor;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Execute;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+@Mojo(name = "rpp", requiresDirectInvocation = true, requiresDependencyResolution = ResolutionScope.TEST)
+@Execute(phase = LifecyclePhase.TEST, lifecycle = "rpp")
+public class RppMojo extends RppHandlerMojo {
+
+    /**
+     * Runs maven surefire.
+     * @throws MojoExecutionException when the surefire invocation fails.
+     */
+    private void invokeSurefire() throws MojoExecutionException {
+        PrintStream stdout = System.out;
+        PrintStream stderr = System.err;
+        try {
+            PrintStream ps = new PrintStream(new FileOutputStream(
+                    getArtifactsDir() + File.separator + "background-surefire-run.txt"));
+            // FIXME: this somehow doesn't redirect the entire surefire output. Debug while PR is ongoing
+            System.setOut(ps);
+            System.setErr(ps);
+            SurefireMojoInterceptor.sfMojo.getClass().getMethod("execute").invoke(SurefireMojoInterceptor.sfMojo);
+            System.setOut(stdout);
+            System.setErr(stderr);
+        } catch (IllegalAccessException ex) {
+            throw new RuntimeException(ex);
+        } catch (InvocationTargetException ex) {
+            throw new RuntimeException(ex);
+        } catch (NoSuchMethodException ex) {
+            throw new RuntimeException(ex);
+        } catch (FileNotFoundException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    /**
+     * Relocates the generated violation-counts file.
+     * @param mode the identifier for the relocated violation-counts file (either "critical" or "background")
+     * @throws MojoExecutionException when obtaining directory information fails.
+     */
+    private void moveViolationCounts(String mode) throws MojoExecutionException {
```

Comment:
This can be something we add in the vms pr but we should move this function to Utils since it gets used by both vms and rpp